tardis/model/tests/test_csvy_model.py

```diff
@@ -34,24 +45,150 @@ def test_compare_models(filename):
     for prop in config_model_props:
         csvy_model_val = csvy_model.get_properties()[prop]
         config_model_val = config_model.get_properties()[prop]
-        if prop == 'homologous_density':
-            npt.assert_array_almost_equal(csvy_model_val.density_0.value, config_model_val.density_0.value)
-            npt.assert_array_almost_equal(csvy_model_val.time_0.value, config_model_val.time_0.value)
+        if prop == "homologous_density":
+            npt.assert_array_almost_equal(
+                csvy_model_val.density_0.value, config_model_val.density_0.value
+            )
+            npt.assert_array_almost_equal(
+                csvy_model_val.time_0.value, config_model_val.time_0.value
+            )
         else:
-            if hasattr(config_model_val, 'value'):
+            if hasattr(config_model_val, "value"):
                 config_model_val = config_model_val.value
                 csvy_model_val = csvy_model_val.value
             npt.assert_array_almost_equal(csvy_model_val, config_model_val)
 
+    assert csvy_model.raw_abundance.shape == config_model.raw_abundance.shape
+    assert (
+        csvy_model.raw_isotope_abundance.shape
+        == config_model.raw_isotope_abundance.shape
+    )
+    assert csvy_model.abundance.shape == config_model.abundance.shape
+    npt.assert_array_almost_equal(
+      csvy_model.raw_abundance.to_numpy(), config_model.raw_abundance.to_numpy()
+    )
+    npt.assert_array_almost_equal(
+        csvy_model.raw_isotope_abundance.to_numpy(),
+        config_model.raw_isotope_abundance.to_numpy(),
+    )
+    npt.assert_array_almost_equal(
+        csvy_model.abundance.to_numpy(), config_model.abundance.to_numpy()
+    )
+
 
-def test_csvy_abundance():
-    csvypath = os.path.join(DATA_PATH, 'config_v_filter.yml')
+@pytest.fixture(scope="module")
+def csvy_model_to_test_abundances():
+    """Returns Radial1DModel to use to test abundances dataframes"""
+    csvypath = os.path.join(DATA_PATH, "csvy_model_to_test_abundances.yml")
     config = Configuration.from_yaml(csvypath)
-    csvy_model = Radial1DModel.from_csvy(config)
-    csvy_abund = csvy_model.abundance
+    csvy_model_to_test_abundances = Radial1DModel.from_csvy(config)
+    return csvy_model_to_test_abundances
+
+
+def test_read_csvy_abundances(csvy_model_to_test_abundances):
+    """Test if model reads abundances and isotope abundances 
+       and constructs dataframes appropriately before applying decay"""
+    #rows in input_abundance are abundances for a fixed element,columns 
+    #represent different shells
+    abundance_index = pd.Index([1, 2], name="atomic_number")
+    input_abundance = pd.DataFrame(
```

Comment:
Would it make sense to move this into a fixture?

Reply:
I did as suggested